File: 11-502/Generics/src/main/java/ru/itis/inform/ArrayList.java
package ru.itis.inform;

import java.lang.reflect.Array;

public class ArrayList<T> {

    private final int MAX_SIZE = 100;

    // ИСПОЛЬЗУЕМ МАССИВ OBJECT
    private Object elements[];

    private int count;
    public ArrayList() {
        elements = new Object[MAX_SIZE];
    }

    public void add(T element) {
        this.elements[count] = element;
        count++;
    }

    @SuppressWarnings("unchecked")
    public T get(int index) {
        if (index >= 0 && index < this.count) {
            return (T)this.elements[index];
        } else throw new IllegalArgumentException();
    }
}

File: 09-551/Lists/ArrayList.cpp
#include "ArrayList.h"
#include <iostream>
using namespace std;

void CreateArrayList(ArrayList *&list)
{
    list = new ArrayList;
    list->array = new int[MAX_SIZE];
    list->count = 0;
}
void AddIntoArrayList(ArrayList *&list, int element)
{
    list->array[list->count] = element;
    list->count++;
}
void Show(ArrayList *list)
{
    for (int i = 0; i < list->count; i++)
    {
        cout << list->array[i] << " ";
    }
}